File: data-structures/linked-list/linked-list-class/python/node.py


class Node:
    def __init__(self, value: str, next=None) -> None:
        self._value = value
        self._next = next

    def get_value(self) -> str:
        return self._value

    def set_value(self, value: str) -> None:
        self._value = value

    def get_next(self) -> object:
        return self._next

    def set_next(self, next: object) -> None:
        self._next = next

File: data-structures/linked-list/linked-list-class/java/Node.java

public class Node {
    private String value;
    private Node next;

    public Node(String value) {
        this.value = value;
        this.next = null;
    }

    public Node(String value, Node next) {
        this.value = value;
        this.next = next;
    }

    public String getValue() {
        return this.value;
    }

    public Node getNext() {
        return this.next;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public void setNext(Node node) {
        this.next = node;
    }
}